composer/algorithms/utils/augmentation_common.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2021 MosaicML. All Rights Reserved.
+from typing import Callable, Iterable, Type, TypeVar, cast
+
+import torch
+from PIL.Image import Image as PillowImage
+from torchvision import transforms
+
+_InputImgT = TypeVar("_InputImgT", torch.Tensor, PillowImage)
+_OutputImgT = TypeVar("_OutputImgT", torch.Tensor, PillowImage)
+
+
+def image_as_type(image: _InputImgT, typ: Type[_OutputImgT]) -> _OutputImgT:
+    """Converts between :class:`torch.Tensor` and
+            :class:`PIL.Image.Image` image representations
+
+    Args:
+        image: a single image represented as a :class:`PIL.Image.Image` or
+            a rank 2 or rank 3 :class:`torch.Tensor` in ``HW`` or ``CHW`` format.
+            A rank 4 or higher tensor can also be provided as long as no type
+            conversion is needed; in this case, the input tensor will be
+            returned. This case is allowed so that functions that natively
+            operate on batch tensors can safely call
+            ``image_as_type(image, torch.Tensor)`` without additional error
+            and type checking.
+        typ: type of the copied image. Must be :class:`PIL.Image.Image`
+            or :class:`torch.Tensor`
+
+    Returns:
+        A copy of ``image`` with type ``typ``
+
+    Raises:
+        TypeError: if ``typ`` is not one of :class:`torch.Tensor` or
+            :class:`PIL.Image.Image`
+        ValueError: if ``image`` cannot be converted to the ``typ``
+
+    """
+    if isinstance(image, typ):
+        return image
+    if not typ in (torch.Tensor, PillowImage):
+        raise TypeError(f"Only typ={{torch.Tensor, Image}} is supported; got {typ}")
+
+    if typ is torch.Tensor:
+        return transforms.functional.to_tensor(image)  # PIL -> Tensor
+    return transforms.functional.to_pil_image(image)  # Tensor -> PIL
```

Comment:
It's possible I missed something in your code or your tests to handle this, but when I run `transforms.functional.to_pil_image(image)` on a rank 4 tensor, I get `ValueError: pic should be 2/3 dimensional. Got 4 dimensions.`

Reply:
yes, it's supposed to fail if it gets a rank 4 tensor. I called this out more explicitly in the Raises section of the docstring